packages/espressocash_app/lib/features/tokens/services/token_service.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:convert';
+
+import 'package:dfunc/dfunc.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:injectable/injectable.dart';
+
+import '../../../data/db/db.dart';
+import '../data/token_repository.dart';
+import '../token.dart';
+
+@lazySingleton
+class TokenService {
+  const TokenService(this.tokenRepository);
+
+  final TokenListRepository tokenRepository;
+
+  Future<Either<Exception, void>> initializeDatabaseFromCsvFile(
+    String filePath,
+  ) =>
+      tryEitherAsync((_) async {
+        final String data = await rootBundle.loadString(filePath);
+        final List<TokenRow> tokenIterable =
+            await compute(_parseAndInsertTokens, data);
+        await tokenRepository.insertTokens(tokenIterable);
```

Comment:
You're still loading the whole (potentially) huge file on the main isolate, push the data to another isolate and then do the saving again on the main isolate. Instead, the whole operation should be done in the isolate, including inserting data into db. Also, preferably, the file itself should be parsed in chunks, not loaded into memory at once.

I would also put thin method directly into repository initialization.